File: algorithms/c/24.c
#include <stdio.h>

struct ListNode
{
    int val;
    struct ListNode *next;
};

struct ListNode* swapPairs(struct ListNode* head)
{
    if ( !head || !head->next )
        return head;
    struct ListNode *p, *q, *t;
    p = head;
    q = head->next;
    t = q->next;
    q->next = p;
    p->next = swapPairs(t);
    return q;
}

int main()
{
    return 0;
}

File: algorithms/python/24.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def swapPairs(self, head):
        """
        :type head: ListNode
        """
        if not head or not head.next:
            return head
        p, q = head, head.next
        q.next, p.next = p, self.swapPairs(q.next)
        return q
